### ISSUE TITLE
Common pattern: conditional within loop

### ISSUE THREAD

**Issue by kytrinyx** [JacobNinja/exercism-analysis#8]
Very often I see people use an enumerable method, and then stick an if/else or ternary statement or a postfix if inside of it. I generally point them to the list of enumerable methods and say that often there will be one that can do more of the heavy lifting.

Is this something that would be fairly low-cost to implement an analyzer for?

**Comment by JacobNinja** [comment]
> Is this something that would be fairly low-cost to implement an analyzer for?

Seems fairly simple to me. Just to clarify: This rule should trigger when a conditional is used within a block called on an `Enumerable` method? Does that include ternary as well?

**Comment by kytrinyx** [comment]
Yeah, ternary as well as `unless` and `if` (both normal and postfix).

**Comment by JacobNinja** [comment]
I have a branch sitting around that needs updated with additional `Enumerable` methods and then it's code complete. I'll get to that this weekend I'm sorry for the delay

**Comment by kytrinyx** [comment]
That's great!
